cv_bridge/include/cv_bridge/cv_mat_sensor_msgs_image_type_adapter.hpp

```diff
@@ -213,15 +253,15 @@ class ROSCvMatContainer
   CV_BRIDGE_PUBLIC
   bool
   is_bigendian() const;
-  
+
   /// Return the encoding override if provided.
   CV_BRIDGE_PUBLIC
   std::optional<std::string>
   encoding_override() const;
 
 private:
   std_msgs::msg::Header header_;
-  cv::Mat frame_;
+  CvImageStorageType frame_;
```

Comment:
Do you think we should pimpl it? In case we need extra data in the future to keep the ABI relatively stable?

Reply:
Not sure how this would keep the ABI more stable, it's not part of the public interface. Can you explain a bit further?

I can understand it for avoiding recompiling issues, but then I have to be more verbose and avoid templates :(  (which tbh is only one function... but still!)

Reply:
Ah get it now. hmm, I don't think we can avoid it. If we add another type we would need to add the signatures to be able to read that type, so I don't think using pimpl would avoid ABI break here :/

Reply:
Sure not with this version, but we are breaking ABI anyway. I rather meant for future changes.

Reply:
I am thinking of people using that type in their packages. Should we add any new members or change them, it will break their ABI without recompilation should the upgrade via apt.